spi/data-plane/data-plane-spi/src/main/java/org/eclipse/edc/connector/dataplane/spi/pipeline/PipelineService.java

```diff
@@ -15,28 +15,14 @@
 package org.eclipse.edc.connector.dataplane.spi.pipeline;
 
 import org.eclipse.edc.runtime.metamodel.annotation.ExtensionPoint;
-import org.eclipse.edc.spi.types.domain.transfer.DataFlowRequest;
-
-import java.util.concurrent.CompletableFuture;
 
 /**
  * Transfers data from a source to a sink.
+ *
+ * @deprecated this interface will be removed from the hierarchy, please use {@link TransferService} instead.
  */
 @ExtensionPoint
+@Deprecated(since = "0.3.0")
```

Comment:
Do we really want to completely get rid of PipelineService? 

In my opinion, it should stay with only the `registerFactory` methods in it. As the `DataSourceFactory` and `DataSinkFactory` will only be used within the PipelineServiceImpl, don't they?

Reply:
I found out that `PipelineService` at this point is exactly the same as `TransferService`, I don't see a reason to keep both, we should either keep one or the other.

Reply:
`TransferService` is also (quoting the javadoc) 
> can satisfy a DataFlowRequest by transferring data from a source to a destination.

so I cannot see the need for having such distinction, also the api is pretty much the same.
What could be a `TransferService` that does not rely on `DataSource` or `DataSink`? Currently there isn't any implementation available.

Reply:
Here is an example: https://github.com/eclipse-edc/Technology-Azure/blob/main/extensions/data-plane/data-plane-azure-data-factory/src/main/java/org/eclipse/edc/connector/dataplane/azure/datafactory/AzureDataFactoryTransferService.java 

`TransferService` was initially introduced when the source/destination pair enables to transfer the data without streaming them through the Data Plane (which is what the `PipelineService` is used for).